client/actions/show-heartbeat/index.js

```diff
@@ -21,11 +30,25 @@ export default class ShowHeartbeatAction extends Action {
     } = this.recipe.arguments;
 
     const lastShown = await this.getLastShownDate();
+
+    // get the last shown for heartbeats in general
+    const lastHeartbeatShown = await this.getLastHeartbeatDate();
+
+    // we can show the heartbeat if there has not been one shown recently,
+    const canShowHeartbeat = lastHeartbeatShown === null
+    // or if the last one shown was more than HEARTBEAT_THROTTLE ms ago
+      || Date.now() - lastHeartbeatShown >= HEARTBEAT_THROTTLE;
```

Comment:
this is kind of weird to me. I dont think we should have comments breaking up boolean expressions.

I'll let @mythmon weigh in too.